extensions/rest/src/lib.rs

```diff
@@ -0,0 +1,182 @@
+            .map(|i| &self.endpoints[i])
+            .ok()
+    }
+
+    pub fn create_filter<'a>(&'a mut self, selection: &str) -> Result<&'a Filter, Error> {
+        if !self.filters.contains_key(selection) {
+            let mut defs = ParseCtx::new(Vec::new());
+
+            let (filter, errors) = jaq_parse::parse(selection, jaq_parse::main());
+
+            if !errors.is_empty() {
+                return Err(Error {
+                    extensions: Vec::new(),
+                    message: format!("The selection is not in valid jq syntax: {errors:?}"),
+                });
+            }
+
+            let filter = defs.compile(filter.expect("we handled errors above"));
+
+            if !defs.errs.is_empty() {
+                return Err(Error {
+                    extensions: Vec::new(),
+                    message: "Error compiling jq filter".to_string(),
+                });
+            }
+
+            self.filters.insert(selection.to_string(), filter);
+        }
+
+        Ok(self.filters.get(selection).unwrap())
+    }
+}
+
+impl Resolver for RestExtension {
+    fn resolve_field(
+        &mut self,
+        _: SharedContext,
+        directive: Directive,
+        _: FieldDefinition,
+        _: FieldInputs,
+    ) -> Result<FieldOutput, Error> {
+        let rest: Rest<'_> = directive.arguments().map_err(|e| Error {
+            extensions: Vec::new(),
+            message: format!("Could not parse directive arguments: {e}"),
+        })?;
+
+        let Some(endpoint) = self.get_endpoint(rest.endpoint) else {
+            return Err(Error {
+                extensions: Vec::new(),
+                message: format!("Endpoint not found: {}", rest.endpoint),
+            });
+        };
+
+        let url = Url::parse(&format!("{}/{}", endpoint.http.base_url, rest.http.path)).map_err(|e| Error {
+            extensions: Vec::new(),
+            message: format!("Could not parse URL: {e}"),
+        })?;
+
+        let request = HttpRequest::builder(url, rest.http.method.into()).build();
+
+        let result = http::execute(&request).map_err(|e| Error {
+            extensions: Vec::new(),
+            message: format!("HTTP request failed: {e}"),
+        })?;
+
+        if !result.status().is_success() {
+            return Err(Error {
+                extensions: Vec::new(),
+                message: format!("HTTP request failed with status: {}", result.status()),
+            });
+        }
+
+        let data: serde_json::Value = result.json().map_err(|e| Error {
+            extensions: Vec::new(),
+            message: format!("Error deserializing response: {e}"),
+        })?;
+
+        let filter = self.create_filter(rest.selection)?;
+        let inputs = RcIter::new(core::iter::empty());
+
+        let filtered = filter.run((Ctx::new([], &inputs), Val::from(data)));
+
+        let mut results = FieldOutput::new();
+
+        for result in filtered {
```

Comment:
That's weird to me,  the number of field outputs must be equal to the number of field inputs. So if there are multiple results, the only thing that makes sense to me is to wrap them in a list and then push it, combined, to the output.